src/parse/decl.rs

```diff
@@ -992,6 +992,11 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
                             next: prefix.map(Box::new),
                         })
                     } else {
+                        if self.match_next(&Token::Keyword(Keyword::Static)).is_some() {
+                            // TODO: Add information for the compiler to know
+                            // about a possible optimization. 
```

Comment:
```suggestion
                            // to warn if `NULL` is passed into the function
```
This is the main benefit I see from `[static n]` declarations.